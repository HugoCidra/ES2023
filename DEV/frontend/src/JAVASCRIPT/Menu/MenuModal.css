/*------------------------------MODAL------------------------------*/

body.active-modal {
    overflow-y: hidden; 
}

.btn-modal {
    padding: 2% 4%;
    display: block;
    margin: 0 auto; 
    font-size: 2rem;
    position: relative;
    z-index: 99; 
}

.overlay {
    background: rgba(255, 255, 255, 0.577);
    z-index: 10; 
    position: fixed;
    width: 100vw;
    height: 100vh;
    top: 0;
    left: 0;
}



.modal {
    width: 240px; 
    position: fixed; 
    z-index: 12; 
    right: 260px; 
    top: 140px; 
    border-radius: 3px; 
    box-shadow: var(--shadow); 
    background-color: white; 
}

.modal-content {
    line-height: 1.4;
    padding: 2% 4%;
    border-radius: 3px; 
    z-index: 13; 

    
}


