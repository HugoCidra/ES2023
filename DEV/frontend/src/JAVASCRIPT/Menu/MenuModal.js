import React from "react";
import "./MenuModal.css";
import Admin from "../Admin/Admin";

export default function MenuModal({ modal, toggleModal }) {


  React.useEffect(() => {
    if (modal) {
      document.body.classList.add('active-modal');
    } else {
      document.body.classList.remove('active-modal');
    }
  }, [modal]); 

  return (
    <>
      {modal && (
        <div className="modal">
          <div onClick={toggleModal} className="overlay"></div>
            <Admin/>
        </div>
      )}
    </>
  );
}