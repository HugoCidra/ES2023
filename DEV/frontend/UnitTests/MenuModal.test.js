import React from 'react';
import { render } from '@testing-library/react';
import MenuModal from '../src/JAVASCRIPT/Menu/MenuModal'; 
import '@testing-library/jest-dom';


describe('MenuModal Component', () => {
  test('MenuModal component renders without crashing', () => {
    render(<MenuModal />);
  });

  test('ContÃªiner do modal possui a classe CSS correta', () => {
    const { container } = render(<MenuModal modal={true} />);
    expect(container.firstChild).toHaveClass('modal');
  });

  test('MenuModal corresponde ao snapshot anterior', () => {
    const { asFragment } = render(<MenuModal />);
    expect(asFragment()).toMatchSnapshot();
  });

});