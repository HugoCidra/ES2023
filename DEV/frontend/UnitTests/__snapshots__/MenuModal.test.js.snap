// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuModal Component MenuModal corresponde ao snapshot anterior 1`] = `<DocumentFragment />`;
